src/keyword.rs

```diff
@@ -138,28 +139,22 @@ impl Model for Keyword {
 
 /// Handles the `GET /keywords` route.
 pub fn index(req: &mut Request) -> CargoResult<Response> {
-    use diesel::expression::dsl::sql;
-    use diesel::types::BigInt;
     use schema::keywords;
 
     let conn = req.db_conn()?;
     let (offset, limit) = req.pagination(10, 100)?;
     let query = req.query();
     let sort = query.get("sort").map(|s| &s[..]).unwrap_or("alpha");
 
-    let mut query = keywords::table
-        .select((keywords::all_columns, sql::<BigInt>("COUNT(*) OVER ()")))
-        .limit(limit)
-        .offset(offset)
-        .into_boxed();
+    let mut query = keywords::table.into_boxed();
 
     if sort == "crates" {
         query = query.order(keywords::crates_cnt.desc());
     } else {
         query = query.order(keywords::keyword.asc());
     }
 
-    let data = query.load::<(Keyword, i64)>(&*conn)?;
+    let data = query.paginate(limit, offset).load::<(Keyword, i64)>(&*conn)?;
```

Comment:
I doubt I'd add it to Diesel, but this could definitely be extracted to a crate. I have considered adding a method that lets you specify the type you're going to deserialize to outside of `load`. (The primary motivation is that `Version::belonging_to(&crate).load::<Version>` bugs me) If that were present, we could definitely have this be `.paginate::<Keyword>` and have the paginate method call that with `(T, i64)`. I haven't explored how much work it'd be to add that method though